File: Restaurant/src/main/java/com/finalproject/Restaurant/model/Restaurant.java
package com.finalproject.Restaurant.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.*;

@Entity
@Table(name = "Restaurant")
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer restaurantId;

    @Column(nullable = false,length = 255)
    private String restaurantName;

    @Column(length = 13)
    private String restaurantPhone;

    @Column(columnDefinition = "TEXT")
    private String restaurantImg;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, length = 50)
    private String latitude;

    @Column(nullable = false, length = 50)
    private String longitude;

    @Column(length = 100)
    private String province;

    @Column(length = 100)
    private String district;

    @Column(length = 100)
    private String subdistrict;

    @Column()
    private LocalDateTime openTime;

    @Column()
    private LocalDateTime closeTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "restaurantTypeId")
    private RestaurantType restaurantType;
}

File: ProjectApplication/lib/model/Restaurant.dart
import 'package:newproject/model/RestaurantType.dart';
import 'package:newproject/utils/time_helper.dart';

class Restaurant {
  int? restaurantId;
  String? restaurantName;
  String? restaurantPhone;
  String? restaurantImg;
  String? description;
  String? latitude;
  String? longitude;
  String? province;
  String? district;
  String? subdistrict;
  String? openTime;  
  String? closeTime; 
  RestaurantType? restaurantType;

  String get formattedOpenTime => formatTime(openTime);
  String get formattedCloseTime => formatTime(closeTime);

  Restaurant({
    this.restaurantId,
    this.restaurantName,
    this.restaurantPhone,
    this.restaurantImg,
    this.description,
    this.latitude,
    this.longitude,
    this.province,
    this.district,
    this.subdistrict,
    this.openTime,
    this.closeTime,
    this.restaurantType,
  });

  factory Restaurant.fromRestaurantJson(Map<String, dynamic> json) {
    return Restaurant(
      restaurantId: json['restaurantId'],
      restaurantName: json['restaurantName'],
      restaurantPhone: json['restaurantPhone'],
      restaurantImg: json['restaurantImg'],
      description: json['description'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      province: json['province'],
      district: json['district'],
      subdistrict: json['subdistrict'],
      openTime: json['openTime'],
      closeTime: json['closeTime'],
      restaurantType: json['restaurantType'] != null
          ? RestaurantType.fromJson(json['restaurantType'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'restaurantId': restaurantId,
      'restaurantName': restaurantName,
      'restaurantPhone': restaurantPhone,
      'restaurantImg': restaurantImg,
      'description': description,
      'latitude': latitude,
      'longitude': longitude,
      'province': province,
      'district': district,
      'subdistrict': subdistrict,
      'openTime': openTime,
      'closeTime': closeTime,
      'restaurantType': restaurantType?.toJson(),
    };
  }
}